### PR TITLE
KAFKA-10787: Apply spotless to transaction-coordinator and server-common

### PR DESCRIPTION
**Pull request by gongxuanzhang**
##  Brief Introduction
This PR is sub PR from https://github.com/apache/kafka/pull/16097.
That Pr we add spotless plugin to build.gradle.
In order not to modify large-scale code, we plan apply it step by step in module. 
The first is 'transaction-coordinator' 'server-common'.
Module  'transaction-coordinator' meets the requirements(only have 2 Java class) so apply it directly.
This PR update 'server-common' module code review import order.
We will apply `spotless` to all module in future

## How to test:
We can run `./gradlew spotlessCheck`  check for code that does not meet requirements. 
If we get report that error , we can run `./gradlew spotlessApply` to review my code.
In this PR, all change(exclude `build.gradle`) ` by `spotlessApply`

### CODE REVIEW

**Review by chia7712 — APPROVED** [review]
LGTM

**Comment by chia7712** [comment]
@gongxuanzhang Could you please update the description to share the tips of "how to test this PR"?

**Comment by gongxuanzhang** [comment]
@chia7712  I changed it , plz take a look

**Comment by chia7712** [comment]
@gongxuanzhang Could you please check the build error?

**Comment by chia7712** [comment]
It seems we are facing Trolley problem :(

1. we use spotless 6.13 in order to **support JDK 8**
2. the google-format version used by spotless 6.13 **has issue in JDK 21** (https://github.com/diffplug/spotless/pull/1920)

---

Given that we are going to get rid of JDK 8 and there is example of ignoring JDK8
https://github.com/apache/kafka/blob/trunk/build.gradle#L2448

Maybe we can remove the support of JDK8 from spotless!

1. upgrade spotless to newest version
2. add `JavaVersion.current().isJava11Compatible()` check to apply spotless only if the env is JDK11+ 

@gongxuanzhang WDYT?

**Comment by gongxuanzhang** [comment]
I agree remove support JDK8.  
I will change spotless version and comment

**Comment by chia7712** [comment]
@gongxuanzhang please fix the conflicts

**Comment by gongxuanzhang** [comment]
@chia7712  complete

File: build.gradle [hunk 1]
```diff
@@ -853,8 +851,8 @@ subprojects {
     skipProjects = [ ":jmh-benchmarks", ":trogdor" ]
     skipConfigurations = [ "zinc" ]
   }
-  
-  if(JavaVersion.current().isJava11Compatible() && project.path !in excludedSpotlessModules) {
+  //  current spotless work error in JDK21,upgrade spotless version when kafka drop support JDK8
```
Comment (chia7712) [inline build.gradle]:
```
the task `removeUnusedImports` is implemented by google-java-format, and unfortunately the google-java-format version used by spotless 6.14.0 can't work with JDK 21. Hence, we apply spotless tasks only if the env is either JDK11 or JDK17
```

WDYT?

**Review by chia7712 — COMMENTED** [review]
@gongxuanzhang Could you please rebase code to trigger QA again?

File: build.gradle (hunk 1 above)
Comment (gongxuanzhang) [reply build.gradle]: i will update comment

**Comment by chia7712** [comment]
I have called QA again

**Comment by chia7712** [comment]
@gongxuanzhang could you please rebase code to include #16249

**Comment by gongxuanzhang** [comment]
Okay, next we wait for CI @chia7712

**Pull request merged**